src/bridge/GasRefunder.sol

```diff
@@ -216,11 +222,20 @@ contract GasRefunder is IGasRefunder, Ownable {
         uint256 maxRefundeeBalance = commonParams.maxRefundeeBalance;
         uint256 maxSingleGasUsage = commonParams.maxSingleGasUsage;
 
+        // add any cost for 4844 data, the data hash reader throws an error prior to 4844 being activated
+        try dataHashReader.getDataHashes() returns (bytes32[] memory dataHashes) {
```

Comment:
I'd put this in the gasUsed calculation of the caller if tx.origin == msg.sender, because as-is a validator might be able to make many calls to their own wallet in one tx while supplying a blob and get refunded for each of them.

Reply:
Yeh, defo safer